src/panels/config/zha/zha-cluster-commands.ts

```diff
@@ -0,0 +1,275 @@
+      this._selectedCommandIndex = -1;
+      this._fetchCommandsForCluster();
+    }
+    super.update(changedProperties);
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      ${this.renderStyle()}
+      <ha-config-section .isWide="${this.isWide}">
+        <div style="position: relative" slot="header">
+          <span>Cluster Commands</span>
+          <paper-icon-button
+            class="toggle-help-icon"
+            @click="${this._onHelpTap}"
+            icon="hass:help-circle"
+          >
+          </paper-icon-button>
+        </div>
+        <span slot="introduction">View and issue cluster commands.</span>
+
+        <paper-card class="content">
+          ${this._renderCommandPicker()}
+          ${
+            this._selectedCommandIndex !== -1
+              ? this._renderCommandInteractions()
+              : ""
+          }
+        </paper-card>
+      </ha-config-section>
+    `;
+  }
+
+  private _renderCommandPicker(): TemplateResult {
+    return html`
+      <div class="command-picker">
+        <paper-dropdown-menu
+          label="Commands of the selected cluster"
+          dynamic-align=""
+          class="flex"
+        >
+          <paper-listbox
+            slot="dropdown-content"
+            selected="${this._selectedCommandIndex}"
+            @iron-select="${this._selectedCommandChanged}"
+          >
+            ${
+              this._commands.map(
+                (entry) => html`
+                  <paper-item
+                    >${entry.name + " (id: " + entry.id + ")"}</paper-item
+                  >
+                `
+              )
+            }
+          </paper-listbox>
+        </paper-dropdown-menu>
+      </div>
+
+      <div ?hidden="${!this._showHelp}" style="color: grey; padding: 16px">
+        Select a command to interact with
+      </div>
+    `;
+  }
+
+  private _renderCommandInteractions(): TemplateResult {
+    return html`
+      <div class="input-text">
+        <paper-input
+          label="Manufacturer code override"
+          type="number"
+          value="${this._manufacturerCodeOverride}"
+          @value-changed="${this._onManufacturerCodeOverrideChanged}"
+          placeholder="Value"
+        ></paper-input>
+      </div>
+      <div class="card-actions">
+        <ha-call-service-button
+          .hass="${this.hass}"
+          domain="zha"
+          service="issue_zigbee_cluster_command"
+          .serviceData="${this._issueClusterCommandServiceData}"
+          >Issue Zigbee Command</ha-call-service-button
+        >
+        <ha-service-description
+          .hass="${this.hass}"
+          domain="zha"
+          service="issue_zigbee_cluster_command"
+          ?hidden="${!this._showHelp}"
+        ></ha-service-description>
+      </div>
+    `;
+  }
+
+  private async _fetchCommandsForCluster(): Promise<any> {
+    if (this.selectedEntity && this.selectedCluster) {
+      this._commands = await this.hass!.callWS({
```

Comment:
This looks to be the same call as the previous one, so once extract to `src/data/zha.ts` we can avoid duplicating the code as well 👍

Reply:
not the same but will move too